File: fetch/FetchDataProcedural.py
class Person:
    def __init__(self, id, name):
        self.id = id
        self.name = name

people = [
    Person(1, "Alice"),
    Person(2, "Bob"),
    Person(3, "Charlie")
]

target_id = 2
target_person = None
for person in people:
    if person.id == target_id:
        target_person = person
        break

if target_person:
    print(f"Found person with ID {target_id}: {target_person.name}")
else:
    print(f"No person found with ID {target_id}")

File: fetch/FetchDataProcedural.ts
class Person {
  constructor(public id: number, public name: string) {}
}

const people: Person[] = [
  new Person(1, "Alice"),
  new Person(2, "Bob"),
  new Person(3, "Charlie"),
];

const targetId = 2;
let targetPerson: Person | undefined;

for (const person of people) {
  if (person.id === targetId) {
    targetPerson = person;
    break;
  }
}

if (targetPerson) {
  console.log(`Found person with ID ${targetId}: ${targetPerson.name}`);
} else {
  console.log(`No person found with ID ${targetId}`);
}
